### PR TITLE
[Don't merge] Testing Static Linked OpenSSL Release on CKB's CI Platform

### PR DESCRIPTION
**Pull request by eval-exec**
### What problem does this PR solve?
It's hard to test CI release build on my local environment.
So I temporarily modified the CI workflows of CKB to use the CKB CI runner for testing the static linked openssl package instructions.
This PR is based on the changes of #4179 

What's Changed:

### Related changes

- temporarily modified the CI workflows of CKB to use the CKB CI runner for testing the static linked openssl package instructions
- Delete all CI workflows unrelated to packaging CKB release.

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code ci-runs-only: [ quick_checks,linters ]

Side effects

- None

### Release note 

```release-note
None: Exclude this PR from the release note.
```

### CODE REVIEW

**Review by doitian — CHANGES_REQUESTED** [review]
Avoid accidental merging

**Comment by eval-exec** [comment]
Successfully built CKB with static liked openssl on macos, but it can't be executed:
```bash
/Users/runner/work/_temp/ea779bb1-a2af-4800-9a99-155b55d275fb.sh: line 8: 61869 Illegal instruction: 4  ./target/prod/ckb --version
Error: Process completed with exit code 132.
```

Investigating why....

File: devtools/ci/ci_main.sh [hunk 1]
```diff
@@ -71,15 +71,15 @@ case $GITHUB_WORKFLOW in
     sudo apt-get install -y gcc-multilib
     sudo apt-get install -y gcc-aarch64-linux-gnu g++-aarch64-linux-gnu clang
     rustup target add aarch64-unknown-linux-gnu
-    curl -LO https://www.openssl.org/source/openssl-1.1.1.tar.gz
-    tar -xvzf openssl-1.1.1.tar.gz
-    cd openssl-1.1.1
+    curl -LO https://www.openssl.org/source/openssl-3.1.3.tar.gz
+    tar -xvzf openssl-3.1.3.tar.gz
+    cd openssl-3.1.3
```
Comment (chenyukang) [inline devtools/ci/ci_main.sh:76]: test

**Comment by eval-exec** [comment]
It's weired, I login to the CI machine, replayed the instructions of "build openssl" and "build ckb with static linked openssl", no error occurred.

**Comment by eval-exec** [comment]
```bash
we build openssl:
OpenSSL 3.1.3 19 Sep 2023 (Library: OpenSSL 3.1.3 19 Sep 2023)
built on: Thu Dec  7 07:33:11 2023 UTC
platform: darwin64-x86_64
options:  bn(64,64)
compiler: cc -fPIC -arch x86_64 -O3 -Wall -DL_ENDIAN -DOPENSSL_PIC -D_REENTRANT -DOPENSSL_BUILDING_OPENSSL -DNDEBUG
OPENSSLDIR: "/Users/runner/work/ckb/ckb/target/openssl-3.1.3/build/ssl"
ENGINESDIR: "/Users/runner/work/ckb/ckb/target/openssl-3.1.3/build/lib/engines-3"
MODULESDIR: "/Users/runner/work/ckb/ckb/target/openssl-3.1.3/build/lib/ossl-modules"
Seeding source: os-specific
CPUINFO: OPENSSL_ia32cap=0xffba222b5f8bffff:0x283
system openssl:
OpenSSL 1.1.1w  11 Sep 2023
built on: Mon Sep 11 14:08:11 2023 UTC
platform: darwin64-x86_64-cc
options:  bn(64,64) rc4(16x,int) des(int) idea(int) blowfish(ptr) 
compiler: clang -fPIC -arch x86_64 -O3 -Wall -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -D_REENTRANT -DNDEBUG
OPENSSLDIR: "/usr/local/etc/openssl@1.1"
ENGINESDIR: "/usr/local/Cellar/openssl@1.1/1.1.1w/lib/engines-1.1"
Seeding source: os-specific
```
There is a diffirence betwenn "I custom compiled openssl 3.1.3" and "brew installed openssl 1.1": `darwin64-x86_64-cc` vs `darwin64-x86_64`

Trying to specify `darwin64-x86_64-cc` as platform param for `./Configure`

**Pull request closed without merging**